test/framework/modules.py

```diff
@@ -1223,17 +1229,66 @@ def test_module_use_unuse(self):
             self.assertFalse('MODULEPATH' in os.environ)
             os.environ['MODULEPATH'] = old_module_path  # Restore
 
-            # Using an empty path still works (technically) (Lmod only, ignored by Tcl)
-            old_module_path = os.environ['MODULEPATH']
-            self.modtool.use('')
-            self.assertEqual(os.environ['MODULEPATH'], ':' + old_module_path)
-            self.modtool.unuse('')
-            self.assertEqual(os.environ['MODULEPATH'], old_module_path)
-            # Even works when the whole path is empty
-            os.environ['MODULEPATH'] = ''
-            self.modtool.unuse('')
-            self.assertFalse('MODULEPATH' in os.environ)
-            os.environ['MODULEPATH'] = old_module_path  # Restore
+    def test_add_and_remove_module_path(self):
+        """Test add_module_path and whether remove_module_path undoes changes of add_module_path"""
+        test_dir1 = tempfile.mkdtemp(suffix="_dir1")
+        test_dir2 = tempfile.mkdtemp(suffix="_dir2")
+        old_module_path = os.environ.get('MODULEPATH')
+        del os.environ['MODULEPATH']
+        self.modtool.add_module_path(test_dir1)
+        self.assertEqual(os.environ['MODULEPATH'], test_dir1)
+        self.modtool.add_module_path(test_dir2)
+        self.assertEqual(os.environ['MODULEPATH'], test_dir2 + ':' + test_dir1)
+        # Adding the same path does not change the path
+        self.modtool.add_module_path(test_dir1)
+        self.assertEqual(os.environ['MODULEPATH'], test_dir2 + ':' + test_dir1)
+        self.modtool.add_module_path(test_dir2)
+        self.assertEqual(os.environ['MODULEPATH'], test_dir2 + ':' + test_dir1)
+        # Even when a (meaningless) slash is added
+        # This occurs when using an empty modules directory name
+        self.modtool.add_module_path(os.path.join(test_dir1, ''))
+        self.assertEqual(os.environ['MODULEPATH'], test_dir2 + ':' + test_dir1)
```

Comment:
These should really use join using os.pathsep for test_dir2 + ':' + test_dir1
Same thing below. And perhaps a var that contains that join instead of repeating it.

Reply:
Makes sense. Done